design/baremetal-operator/bmo-branching.md

```diff
@@ -0,0 +1,25 @@
+# BMO Branching Proposal
+
+Necessity of release and development branches have been acknowledged in metal3-io/baremetal-operator
+repository lately and the community has decided to introduce the branching as well as updating the release policy
+to maintain the branches. The issue was raised in <https://github.com/metal3-io/baremetal-operator/issues/1283> and
+an update was made to BMO release document <https://github.com/metal3-io/baremetal-operator/pull/1298> as a result.
+
+As a next step, the following proposal is made for the branching:
+
+- BMO e2e (completely independent of CAPM3) test has to be designed and put in place. So far BMO PRs are tested with CAPM3, CAPI and IPAM integration tests but with the introduction of BMO release branches and proper releases from those branches, BMO should not rely on CAPM3 releases or should be decoupled with CAPM3 integration when testing a BMO PR. We propose a completely independent BMO e2e test framework which will test BMO features with ironic. For this few action points have been identified:
+
+   - Simplify BMO deployment through kustomize and get rid of deploy.sh script in the process.
+
+   - Introduce BMO e2e test framework similar to what we have currently in CAPM3.
+
+   - Write all the e2e feature tests necessary to test BMO code with ironic deployment which would be necessary to test integration of a PR landing in BMO.
+
+- Introduce nightly jobs where we would test CAPM3 main branch integration with BMO main branch. This would help us identify if we need to introduce any change in CAPM3/BMO regarding the changes landing in BMO/CAPM3 repo.  Action points:
+
+   - JJBs for nightly fullstack jobs need to be introduced.
+   - Introduce necessary changes in metal3-io/project-infra and metal3-io/metal3-dev-env to accomodate the full stack jobs
```

Comment:
Are not the jobs we have for CAPM3 already enough? Or are they not for BMO main branch?

Reply:
Actually yes, the main branch test for CAPM3 is already testing with main branch for BMO, I am now thinking would we need similar jobs for CAPM3-BMO release combinations once we have different BMO release branches?

Reply:
I think the simplest would be to lock CAPM3 minor releases to BMO minor releases. So for example the CAPM3 release-1.4 test would run with BMO release-x.y. Then when we release CAPM3 v1.5 we can uplift BMO if there is a new release available and so on.

Reply:
What @lentzi90 said is what we agreed on. We are currently using exact tags of those releases, and we would instead use release branches.